percona-xtradb-cluster-8.0-backup/lib/pxc/backup.sh

```diff
@@ -29,8 +29,47 @@ log() {
 clean_backup_s3() {
 	s3_add_bucket_dest
 
-	is_object_exist "$S3_BUCKET" "$S3_BUCKET_PATH.$SST_INFO_NAME" || xbcloud delete ${XBCLOUD_ARGS} --storage=s3 --s3-bucket="$S3_BUCKET" "$S3_BUCKET_PATH.$SST_INFO_NAME"
-	is_object_exist "$S3_BUCKET" "$S3_BUCKET_PATH/" || xbcloud delete ${XBCLOUD_ARGS} --storage=s3 --s3-bucket="$S3_BUCKET" "$S3_BUCKET_PATH"
+	local time=15
+	local is_deleted_full=0
+	local is_deleted_info=0
+	local exit_code=0
+
+	for i in {1..5}; do
+		if (( i > 1 )); then
+			log 'INFO' "Sleeping ${time}s before retry $i..."
+			sleep "$time"
+		fi
+
+		set +e
+		is_object_exist "$S3_BUCKET" "$S3_BUCKET_PATH/"
+		exit_code=$?
+		set -e
+		if [[ $exit_code -ge 1 ]]; then
```

Comment:
we can do this with `if` so we don't need to disable and enable errexit
```suggestion
		if ! is_object_exist "$S3_BUCKET" "$S3_BUCKET_PATH/"; then
```